sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py

```diff
@@ -426,74 +432,81 @@ async def negotiate(self):
+        username, password = None, None
+        http_proxy_host, http_proxy_port = None, None
+        http_proxy_auth = None
+
         if self._http_proxy:
             http_proxy_host = self._http_proxy["proxy_hostname"]
             http_proxy_port = self._http_proxy["proxy_port"]
+            if http_proxy_host and http_proxy_port:
+                http_proxy_host = f"{http_proxy_host}:{http_proxy_port}"
             username = self._http_proxy.get("username", None)
             password = self._http_proxy.get("password", None)
-            if username or password:
-                http_proxy_auth = (username, password)
+
         try:
-            from websocket import create_connection
+            from aiohttp import ClientSession
 
-            self.ws = create_connection(
+            if username or password:
+                from aiohttp import BasicAuth
+                http_proxy_auth = BasicAuth(login=username, password=password)
+                
+            self.session = ClientSession()
+            self.ws = await self.session.ws_connect(
                 url="wss://{}".format(self._custom_endpoint or self.host),
-                subprotocols=[AMQP_WS_SUBPROTOCOL],
                 timeout=self._connect_timeout,
-                skip_utf8_validation=True,
-                sslopt=self.sslopts,
-                http_proxy_host=http_proxy_host,
-                http_proxy_port=http_proxy_port,
-                http_proxy_auth=http_proxy_auth,
+                protocols=[AMQP_WS_SUBPROTOCOL],
+                autoclose=False,
+                proxy=http_proxy_host,
+                proxy_auth=http_proxy_auth,
+                ssl=self.sslopts,
             )
+           
         except ImportError:
-            raise ValueError("Please install websocket-client library to use websocket transport.")
+            raise ValueError("Please install aiohttp library to use websocket transport.")
 
     async def _read(self, n, buffer=None, **kwargs):  # pylint: disable=unused-arguments
         """Read exactly n bytes from the peer."""
-        from websocket import WebSocketTimeoutException
 
         length = 0
         view = buffer or memoryview(bytearray(n))
         nbytes = self._read_buffer.readinto(view)
         length += nbytes
         n -= nbytes
+        
         try:
             while n:
-                data = await self.loop.run_in_executor(None, self.ws.recv)
-
+                data = await self.ws.receive_bytes()
                 if len(data) <= n:
                     view[length : length + len(data)] = data
                     n -= len(data)
                 else:
                     view[length : length + n] = data[0:n]
                     self._read_buffer = BytesIO(data[n:])
-                    n = 0
-
+                    n = 0 
             return view
-        except WebSocketTimeoutException:
+        except (asyncio.TimeoutError) as wex:
```

Comment:
Just curious, is there any interesting info in asyncio.TimeoutError that the user might want to see? Like it's internal message? Looks like you throw it away.